### ISSUE TITLE
KeyError in `summarizer.py`

### ISSUE THREAD

**Issue by nick-youngblut** [merenlab/anvio#793]
I'm running into a KeyError when running `anvi-summarize`, and I can't figure out what's going wrong. The error is:

```
Contigs DB ...................................: Initialized: tests/output_amy/anvio/coassemble.db (v. 10)

WARNING
===============================================
ProfileSuperClass found a collection focus, which means it will be initialized
using only the splits in the profile database that are affiliated with the
collection DAS_Tool and all bins it describes.

Auxiliary Data ...............................: Found: tests/output_amy/anvio/merged_profiles/AUXILIARY-DATA.db (v. 2)
Profile Super ................................: Initialized with 1284 of 3168 splits: tests/output_amy/anvio/merged_profiles/PROFILE.db (v. 23)
Taxonomy .....................................: Initiated for taxonomic level for "t_genus"
[24 Mar 18 16:14:11 [Processing "metabat2_low_PE_009" (1 of 5)]] Storing profile data ...                                                                                            Traceback (most recent call last):
  File "/ebio/abt3_projects/software/dev/llmga/.snakemake/conda/083cdc7f/bin/anvi-summarize", line 96, in <module>
    main(args)
  File "/ebio/abt3_projects/software/dev/llmga/.snakemake/conda/083cdc7f/bin/anvi-summarize", line 60, in main
    summary.process()
  File "/ebio/abt3_projects/software/dev/llmga/.snakemake/conda/083cdc7f/lib/python3.5/site-packages/anvio/summarizer.py", line 848, in process
    self.summary['collection'][bin_id]['color'] = self.bins_info_dict[bin_id]['html_color'] or '#212121'
KeyError: 'metabat2_low_PE_009'
```

The workflow leading up to the error includes:

1. coassembly with `megahit`
1. binning with `metabat2`, `maxbin2`, etc.
1. bin selection with `das-tool`
1. `anvi-gen-contigs-database`
1. `anvi-run-hmms`
1. `anvi-run-ncbi-cogs`
1. `anvi-get-dna-sequences-for-gene-calls` + `centrifuge` run
1. `anvi-import-taxonomy`
1. `anvi-profile`
1. `anvi-merge`
1. `anvi-import-collection` (for `metabat`, `maxbin`, etc...., `das_tool`)
1. `anvi-summarize` (summarizing for "das_tool") 

Any idea what's going wrong? I can provide more details if needed. 

```
$ anvi-self-test --version
Anvi'o version ...............................: 4
Profile DB version ...........................: 23
Contigs DB version ...........................: 10
Pan DB version ...............................: 8
Genome data storage version ..................: 6
Auxiliary data storage version ...............: 2

$ lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 16.04.4 LTS
Release:	16.04
Codename:	xenial
```

**Comment by meren** [comment]
I'm on this.

**Comment by meren** [comment]
Despite all my efforts, I couldn't reproduce this :(

Can you please send the exact command for `anvi-import-collection`? Did you use `--bins-info` flag?

If you don't mind sharing with me privately your gzipped contigs and profile db, along with your `anvi-summarize` command, I can try to reproduce and fix this problem. This is not something we run into before :/

**Comment by nick-youngblut** [comment]
My `anvi-import-collection` command consists of basically the following:

```
anvi-import-collection -p profile.db -c contigs.db -C "DAS_Tool"  --contigs-mode bins.tsv
```

I'm happy to share the files, but preferably privately. Just let me know a private file transfer method that you prefer.

**Comment by meren** [comment]
You can upload them anywhere you like (i.e., Dropbox, Google Drive, etc), and send me a private e-mail with download links and instructions re: which commands to run (so I exactly follow your footsteps).

Thanks!

**Comment by nick-youngblut** [comment]
This the issue fixed or is there just a better error message? I'm still getting the error with other datasets. Example:

```
  File "/ebio/abt3_projects/vadinCA11/bin/llmga/.snakemake/conda/c6ca6cb0/bin/anvi-summarize", line 9
    main(args)
  File "/ebio/abt3_projects/vadinCA11/bin/llmga/.snakemake/conda/c6ca6cb0/bin/anvi-summarize", line 6
    summary.process()
  File "/ebio/abt3_projects/vadinCA11/bin/llmga/.snakemake/conda/c6ca6cb0/lib/python3.5/site-packages
    self.summary['collection'][bin_id]['color'] = self.bins_info_dict[bin_id]['html_color'] or '#2121
KeyError: 'metabat2_high_PE_237'
```

Any idea when this bugfix will be available via bioconda?

**Comment by meren** [comment]
Hi Nick,

The error was due to quite a core bug very deep down in the design. The problem that causes this was a by product of something that is now fixed in commits that impact #795.

There will be no better error message for this version :) If you switch to the `master` branch, things should work fine. A stable version that will appear in bioconda may take some time (say weeks?). But I will talk to @ozcan about this, since we did fix multiple critical things, including #759, and maybe we can think about a quick intermediate release.

Best,
